restapi/user_objects.go

```diff
@@ -305,26 +317,73 @@ func listBucketObjects(ctx context.Context, client MinioClient, bucketName strin
 func getDownloadObjectResponse(session *models.Principal, params user_api.DownloadObjectParams) (io.ReadCloser, *models.Error) {
 	ctx := context.Background()
 	var prefix string
-	if params.Prefix != "" {
+	if *params.IsFolder {
 		encodedPrefix := SanitizeEncodedPrefix(params.Prefix)
 		decodedPrefix, err := base64.StdEncoding.DecodeString(encodedPrefix)
 		if err != nil {
 			return nil, prepareError(err)
 		}
 		prefix = string(decodedPrefix)
+		mClient, err := newMinioClient(session)
+		if err != nil {
+			return nil, prepareError(err)
+		}
+		minioClient := minioClient{client: mClient}
+		objects, err := listBucketObjects(ctx, minioClient, params.BucketName, prefix, true, false, false)
+		if err != nil {
+			return nil, prepareError(err)
+		}
+		var folder string
+		folders := strings.Split(prefix, "/")
+		if len(folders) > 1 {
+			folder = folders[len(folders)-2]
+		}
+		fmt.Println(folders)
+		w := new(bytes.Buffer)
+		zipw := zip.NewWriter(w)
+		for i := 0; i < len(objects); i++ {
+			name := folder + objects[i].Name[len(prefix)-1:]
+			s3Client, err := newS3BucketClient(session, params.BucketName, objects[i].Name)
+			if err != nil {
+				return nil, prepareError(err)
+			}
+			mcClient := mcClient{client: s3Client}
+			object, err := downloadObject(ctx, mcClient, params.VersionID)
+			if err != nil {
+				return nil, prepareError(err)
+			}
+			f, err := zipw.Create(name)
+			if err == nil {
+				buf := new(bytes.Buffer)
+				buf.ReadFrom(object)
+				f.Write(buf.Bytes())
+			}
```

Comment:
Validate the other way around, currently your code is swallowing errors

```suggestion
			if err != nil {
                             return nil, prepareError(err)
			}
			buf := new(bytes.Buffer)
			buf.ReadFrom(object)
			f.Write(buf.Bytes())
```